packages/components/src/internal/AssayDefinitionModel.ts

```diff
@@ -297,4 +297,11 @@ export class AssayDefinitionModel extends ImmutableRecord({
 
         return columns.asImmutable();
     }
+
+    domainFileColumnNames(type: AssayDomainTypes): string[] {
+        return this.getDomainByType(type)
```

Comment:
`this.getDomainByType(type)` can return `undefined`. Add defensive checks here and return an empty array or undefined.